# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: ipc/chromium/src/base/pickle.h
# Commit: 451508713a62
# Full Hash: 451508713a625beaaa9fcbbf06c65ca312dc34fb
# Author: Nika Layzell <nika@thelayzells.com>
# Date: 2021-06-23 09:53:24
# Regressor Bug: 1706374
# File Overlap Count: 2
# Description:
#   Bug 1706374 - Part 13: Remove the event footer from an IPC::Message when deserializing, r=handyman,glandium
#   
#   This unfortunately requires a new method to be added to BufferList to
#   support truncating the buffer to a particular iterator.
#   
# ==============================================================================

diff -r 878722e87e62 -r 451508713a62 ipc/chromium/src/base/pickle.h
--- a/ipc/chromium/src/base/pickle.h	Tue Jun 22 18:17:23 2021 +0000
+++ b/ipc/chromium/src/base/pickle.h	Tue Jun 22 18:17:23 2021 +0000
@@ -154,6 +154,10 @@
   // will succeed.
   bool HasBytesAvailable(const PickleIterator* iter, uint32_t len) const;
 
+  // Truncate the message at the current point, discarding any data after this
+  // point in the message.
+  void Truncate(PickleIterator* iter);
+
   // Methods for adding to the payload of the Pickle.  These values are
   // appended to the end of the Pickle's payload.  When reading values from a
   // Pickle, it is important to read them in the order in which they were added